pkg/aws/awsapi.go

```diff
@@ -0,0 +1,68 @@
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/pricing"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+)
+
+type EC2API interface {
```

Comment:
Putting this interface in the same top-level package is going to cause circular dependencies. I think this needs to be thrown in a different package